Validate address

angaither opened the pull request:
- Add system settings form 
- Valid XML according the UPS guidelines
- Makes post request to UPS api & print the results.

angaither committed:
Intial pass at address validation
- create menu callback for an ajax request
- format XML request

Refs #74

angaither committed:
Added a system settings form for ups info.

aaronschachter commented on lib/modules/dosomething/dosomething_user/dosomething_user.module:
```diff
@@ -29,6 +30,46 @@ function dosomething_user_init() {
+    'page arguments' => array('dosomething_user_ups_api_settings'),
+    'file' => 'dosomething_user_valid_address.inc',
+    'access callback' => 'user_access',
+    'access arguments' => array('administer UPS address vsalidation'),
```
spelling.

aaronschachter commented on lib/modules/dosomething/dosomething_user/dosomething_user.module:
```diff
@@ -29,6 +30,46 @@ function dosomething_user_init() {
+ */
+function dosomething_user_permission() {
+  return array(
+    'administer UPS address vsalidation' =>  array(
```
spelling.

angaither committed:
Fix typo

aaronschachter commented on lib/modules/dosomething/dosomething_user/dosomething_user_valid_address.inc:
```diff
@@ -0,0 +1,70 @@
+  $form['dosomething_access_license_number'] = array(
+    '#type' => 'textfield',
+    '#title' => t('UPS access license number'),
+    '#default_value' => variable_get('dosomething_access_license_number'),
```
Why not store this variable with the same prefix, `dosomething_ups` .. ?  Or additionally use the `dosomething_user` module prefix as well to better specify where its being set?

angaither commented:
@aaronschachter at least I'm consistent.  :smile:

angaither committed:
Updated var names

angaither committed:
Added a dev flag to the ups form

aaronschachter commented on lib/modules/dosomething/dosomething_user/dosomething_user_valid_address.inc:
```diff
@@ -0,0 +1,76 @@
+
+
+/**
+ * Compiles XML according to USPS standards.
```
Is all of this new code for the USPS or UPS?  If USPS, please update the variable and permission names accordingly (or fix this comment to standardize)

angaither replied:
The fake xml stuff was originally for UPS, will be updated in a new PR.

aaronschachter commented on lib/modules/dosomething/dosomething_user/dosomething_user_valid_address.inc:
```diff
@@ -0,0 +1,76 @@
+/**
+ * Compiles XML according to USPS standards.
+ */
+function dosomething_user_compile_xml($address) {
```
`compile_xml` sounds very generic, like it should belong in `dosomething_helpers`.  Perhaps rename to dosomething like `dosomething_user_compile_usps_xml` ?

angaither replied:
same thing.

angaither committed:
Updated xml request to UPS api.
Formats all XML as UPS defines.
Return the results from the api call to the page.

aaronschachter commented on lib/modules/dosomething/dosomething_user/dosomething_user.module:
```diff
@@ -29,6 +30,46 @@ function dosomething_user_init() {
+    'file' => 'dosomething_user_valid_address.inc'
+  );
+
+  $items['admin/config/services/ups-api-settings'] = array(
```
Same question, all this code is for USPS, not UPS right?  Can we do a global replace for anything UPS as USPS if so?

angaither replied:
It's all for UPS, don't think there's anything referencing USPS anymore.

aaronschachter replied:
Sweet.

aaronschachter commented on lib/modules/dosomething/dosomething_user/dosomething_user_valid_address.inc:
```diff
@@ -0,0 +1,203 @@
+ */
+function dosomething_user_validate_address($uid) {
+  $user = user_load($uid);
+  $address = $user->field_address[LANGUAGE_NONE][0];
```
These will cause notices if there's no value for address / firstname.

angaither replied:
Yes, however since this is going to be an ajax callback on the address before saving to the user, this isn't how we will be getting the data. Still need to talk to the fenders about how to do this in the modal...

aaronschachter commented on lib/modules/dosomething/dosomething_user/dosomething_user_valid_address.inc:
```diff
@@ -0,0 +1,203 @@
+function dosomething_user_validate_address($uid) {
+  $user = user_load($uid);
+  $address = $user->field_address[LANGUAGE_NONE][0];
+  $first_name = $user->field_first_name[LANGUAGE_NONE][0]['value'];
```
Same here.
